src/io/sendmsg_zc.rs

```diff
@@ -0,0 +1,107 @@
+use crate::io::SharedFd;
+use crate::runtime::driver::op::{Completable, CqeResult, MultiCQEFuture, Op, Updateable};
+use crate::runtime::CONTEXT;
+use socket2::SockAddr;
+use std::io;
+use std::io::IoSlice;
+use std::net::SocketAddr;
+
+pub(crate) struct SendMsgZc {
+    #[allow(dead_code)]
+    fd: SharedFd,
+    #[allow(dead_code)]
+    io_slices: Vec<IoSlice<'static>>,
+    #[allow(dead_code)]
+    socket_addr: Box<SockAddr>,
+    msg_control: Option<IoSlice<'static>>,
+    pub(crate) msghdr: libc::msghdr,
+
+    /// Hold the number of transmitted bytes
+    bytes: usize,
+}
+
+impl Op<SendMsgZc, MultiCQEFuture> {
+    pub(crate) fn sendmsg_zc(
+        fd: &SharedFd,
+        io_slices: Vec<IoSlice<'static>>,
```

Comment:
I'm still not sure this is right @FrankReh, @MA-ETL . Would a `Vec<T: IoBuf>` be a better choice here?

Would that not allow more flexibility in constructing a send from non-contiguous buffers? I would take ownership of the buffers, as the current ops do, and construct a `msghdr` in place, like now. The user is then freed from buffer handling, and the dreaded `static` constraint is banished

It may also open the door to compatibility with `FixedBuffers`, although I'm unsure if that implemented in io_uring yet.

Reply:
That seems like a good point. Sorry to @MA-ETL that I didn't think of that in my first replay. Thank you @ollie-etl .